assets/src/edit-story/components/library/libraryTabs.js

```diff
@@ -14,30 +14,55 @@
  * limitations under the License.
  */
 
+/**
+ * WordPress dependencies
+ */
+import { useRef, useCallback } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
+import { useKeyDownEffect } from '../keyboard';
 import useLibrary from './useLibrary';
-import { Tabs, Media, Text, Shapes, Links } from './tabs';
+import { Tabs, getPanes } from './panes';
+
+function getTabId(tab) {
+  return `#library-tab-${tab}`;
+}
 
 function LibraryTabs() {
   const {
     state: { tab },
     actions: { setTab },
-    data: {
-      tabs: { MEDIA, TEXT, SHAPES, LINKS },
-    },
+    data: { tabs },
   } = useLibrary();
-  const tabs = [
-    [MEDIA, Media],
-    [TEXT, Text],
-    [SHAPES, Shapes],
-    [LINKS, Links],
-  ];
+  const panes = getPanes(tabs);
+  const ref = useRef();
+  const handleNavigation = useCallback(
+    (isPrevious) => () => {
+      const currentPane = panes.find(({ id }) => id === tab);
+      const nextTab = isPrevious ? currentPane.previous : currentPane.next;
+      if (!nextTab) {
+        return;
+      }
+
+      setTab(nextTab);
+      ref.current.querySelector(getTabId(nextTab)).focus();
+    },
+    [tab, setTab, panes]
+  );
+  // todo: support RTL
+  useKeyDownEffect(ref, 'left', handleNavigation(true), [tab, setTab, panes]);
+  useKeyDownEffect(ref, 'right', handleNavigation(false), [tab, setTab, panes]);
```

Comment:
This was soooo easy to do! Thanks, @dvoytenko! And because of the DOM based model, I don't even need to add anything about "only if focus", because key events will only get to the parent node, if current focus is on one of the tabs.

I don't even see a need to generalize this for other components needing similar but different implementation, as it's so easy.

I didn't find a good way to make it a single callback though. I first examined `evt.key`, but its value is `ArrowLeft` which would be a different syntax than the simple `left` used in setting up the callback. Which wouldI found quite confusing. Is there a way to catch which key has been pressed in MouseTrap lingo? However, using different callbacks works well, so it's not a big issue here.

Reply:
Unfortunately, only partially possible, I believe. I took a look superficially before. It looks like handler passed to `bind` takes two args: KeyboardEvent and what they call `combo`. The `combo` uses their naming, but it contains modifiers and everything. E.g. it'd be `mod+down`. But I could be wrong.

Reply:
Bottom line: I figured in some other places, I could live with dichotomy for a few special cases such as arrows.

Reply:
I'll leave it as is for now.